tests/e2e/scripts/run_e2e_test.sh

```diff
@@ -1,44 +1,125 @@
 #!/usr/bin/env bash
 set -ex
 
-source $(cd $(dirname "$0"); pwd)/util
+# Usage: run_e2e_test.sh <--no_run_lcp> <--zkdcap|--mock_zkdcap> <--enclave_debug> <--upgrade_test> <--key_expiration=<integer>>
 
-export LCP_ENCLAVE_DEBUG=1
+source $(cd $(dirname "$0"); pwd)/util
 
-ENCLAVE_PATH=./bin/enclave.signed.so
-LCP_BIN=${LCP_BIN:-lcp}
+E2E_TEST_DIR=./tests/e2e/cases/tm2eth
+export NO_RUN_LCP=false
+export LCP_ENCLAVE_DEBUG=0
+export LCP_KEY_EXPIRATION=86400
+# LCP_RISC0_IMAGE_ID must be set to the same value as in the LCP service
+LCP_RISC0_IMAGE_ID=${LCP_RISC0_IMAGE_ID:-0xa5b1925a6471aa2d53847816b8c7ec5d09ff455583a4ae66915c402a3216029e}
+export ZKDCAP=false
+export LCP_ZKDCAP_RISC0_MOCK=false
+export LCP_RISC0_IMAGE_ID
+export USE_UPGRADE_TEST=no
 CERTS_DIR=./tests/certs
+ARGS=$(getopt -o '' --long no_run_lcp,enclave_debug,zkdcap,mock_zkdcap,upgrade_test,key_expiration: -- "$@")
+eval set -- "$ARGS"
+while true; do
+    case "$1" in
+        --no_run_lcp)
+            echo "Skip running LCP"
+            NO_RUN_LCP=true
+            shift
+            ;;
+        --enclave_debug)
+            echo "Enclave debug enabled"
+            LCP_ENCLAVE_DEBUG=1
+            shift
+            ;;
+        --zkdcap)
+            echo "ZKDCAP enabled"
+            ZKDCAP=true
+            LCP_ZKDCAP_RISC0_MOCK=false
+            shift
+            ;;
+        --mock_zkdcap)
+            echo "Mock ZKDCAP enabled"
+            ZKDCAP=true
+            LCP_ZKDCAP_RISC0_MOCK=true
+            shift
+            ;;
+        --key_expiration)
+            echo "Key expiration set to $2"
+            LCP_KEY_EXPIRATION=$2
+            shift 2
+            ;;
+        --upgrade_test)
+            echo "Enable upgrade test"
+            USE_UPGRADE_TEST=yes
+            shift
+            ;;
+        --)
+            shift
+            break
+            ;;
+        *)
+            echo "Internal error!"
+            exit 1
+            ;;
+    esac
+done
 
-./tests/e2e/scripts/init_lcp.sh
-
-if [ "$SGX_MODE" = "SW" ]; then
-    export LCP_RA_ROOT_CERT_HEX=$(cat ${CERTS_DIR}/root.crt | xxd -p -c 1000000)
+if [ "$NO_RUN_LCP" = "false" ]; then
+    echo "Run LCP for testing"
+    LCP_BIN=${LCP_BIN:-./bin/lcp}
+    LCP_ENCLAVE_PATH=${LCP_ENCLAVE_PATH:-./bin/enclave.signed.so}
+    export LCP_MRENCLAVE=$(${LCP_BIN} enclave metadata --enclave=${LCP_ENCLAVE_PATH} | jq -r .mrenclave)
+    LCP_BIN=${LCP_BIN} LCP_ENCLAVE_PATH=${LCP_ENCLAVE_PATH} ./tests/e2e/scripts/init_lcp.sh
+    ${LCP_BIN} --log_level=info service start --enclave=${LCP_ENCLAVE_PATH} --address=127.0.0.1:50051 --threads=2 &
+    LCP_PID=$!
+    if [ "$SGX_MODE" = "SW" ]; then
+        export LCP_RA_ROOT_CERT_HEX=$(cat ${CERTS_DIR}/root.crt | xxd -p -c 1000000)
+        export LCP_DCAP_RA_ROOT_CERT_HEX=$(cat ${CERTS_DIR}/simulate_dcap_root_cert.pem | xxd -p -c 1000000)
+    fi
```

Comment:
What happens if `$SGX_MODE = HW`?
In the case, `LCP_RA_ROOT_CERT_HEX` and `LCP_DCAP_RA_ROOT_CERT_HEX` would not be set.
Is it still OK?

Reply:
Yes, no problem. If these env vars are not set, simapp uses the intel root certs as default: https://github.com/datachainlab/lcp-go/blob/101ba8c9dfa1b7c9153490dfd91c5242d9e43862/sgx/dcap/cert_env.go#L13